src/mac_features/ack_generator/nrf_802154_ack_data.h

```diff
@@ -99,6 +101,18 @@ bool nrf_802154_ack_data_for_addr_clear(const uint8_t * p_addr, bool extended, u
  */
 void nrf_802154_ack_data_reset(bool extended, uint8_t data_type);
 
+/**
+ * @brief Select the source matching algorithm.
+ *
+ * @note This method should only be called in the initialization phase.
+ *
+ * When calling @ref nrf_802154_ack_data_pending_bit_should_be_set method, several different
```

Comment:
Imprecise comment. It suggests that the algorithm is a parameter that can be sent to the `should_be_set` function. Please refactor the comment being explicit, e.g. (just an idea):

```
This method can be used to choose the algorithm that determines whether the pending bit should be set.
In order for @ref nrf_802154_ack_data_pending_bit_should_be_set to use the specified algorithm,
this function should be called beforehand, upon driver initialization.
```